cursive/src/backends/termion.rs

```diff
@@ -40,8 +39,59 @@ pub struct Backend {
     // Inner state required to parse input
     last_button: Option<MouseButton>,
 
-    input_receiver: Receiver<TEvent>,
+    events: Events<File>,
     resize_receiver: Receiver<()>,
+    running: Arc<AtomicBool>,
+}
+
+/// Set the given file to be read in non-blocking mode. That is, attempting a
+/// read on the given file may return 0 bytes.
+///
+/// Copied from private function at https://docs.rs/nonblock/0.1.0/nonblock/.
+///
+/// The MIT License (MIT)
+///
+/// Copyright (c) 2016 Anthony Nowell
+///
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// The above copyright notice and this permission notice shall be included in all
+/// copies or substantial portions of the Software.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+/// SOFTWARE.
+#[cfg(unix)]
+fn set_blocking(file: &mut File, blocking: bool) -> std::io::Result<()> {
```

Comment:
Nit: do we need `&mut` here? `AsRawFd` gives `fn as_raw_fd(&self) -> RawFd`. I think it might work with `&File`. This way you also don't need to make `input_file: File` mut in `init_with_files`.

Reply:
Removed `mut`.